pkg/apis/core/validation/validation_test.go

```diff
@@ -20962,6 +21091,61 @@ func TestCrossNamespaceSource(t *testing.T) {
 	}
 }
 
+func pvcSpecWithVolumeAttributesClassName(vacName *string) *core.PersistentVolumeClaimSpec {
+	scName := "csi-plugin"
+	spec := core.PersistentVolumeClaimSpec{
+		AccessModes: []core.PersistentVolumeAccessMode{
+			core.ReadOnlyMany,
+		},
+		Resources: core.ResourceRequirements{
+			Requests: core.ResourceList{
+				core.ResourceName(core.ResourceStorage): resource.MustParse("10G"),
+			},
+		},
+		StorageClassName:          &scName,
+		VolumeAttributesClassName: vacName,
+	}
+	return &spec
+}
+
+func TestVolumeAttributesClass(t *testing.T) {
+	testCases := []struct {
+		testName     string
+		expectedFail bool
+		claimSpec    *core.PersistentVolumeClaimSpec
+	}{
+		{
+			testName:     "Feature gate enabled and valid no volumeAttributesClassName specified",
+			expectedFail: false,
+			claimSpec:    pvcSpecWithVolumeAttributesClassName(nil),
+		},
+		{
+			testName:     "Feature gate enabled and valid volumeAttributesClassName specified",
+			expectedFail: false,
+			claimSpec:    pvcSpecWithVolumeAttributesClassName(utilpointer.String("foo")),
```

Comment:
Can you add a test for "Feature gate disabled and valid volumeAttributesClassName specified"

Reply:
Is it expected to fail or not in this case?

Reply:
If the feature is disabled, volumeAttributesClassName cannot be updated, so the test should fail regardless of what is the volumeAttributesClassName.